core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -688,10 +688,11 @@ private[spark] class TaskSchedulerImpl(
             val errorMsg =
               s"Fail resource offers for barrier stage ${taskSet.stageId} because only " +
                 s"${addressesWithDescs.size} out of a total number of ${taskSet.numTasks}" +
-                s" tasks got resource offers. This happens because barrier execution currently " +
-                s"does not work gracefully with delay scheduling. We highly recommend you to " +
-                s"disable delay scheduling by setting spark.locality.wait=0 as a workaround if " +
-                s"you see this error frequently."
+                s" tasks got resource offers. This could happen if delay scheduling or " +
+                s"blacklisting is enabled, as barrier execution currently does not work " +
+                s"gracefully with them. We highly recommend you to disable delay scheduling " +
+                s"by setting spark.locality.wait=0 or disable blacklisting by setting " +
```

Comment:
I'm not sure if we should recommend disabling blacklisting.  If it was supposed to be blacklisted assumption is something was wrong with it so job would probably fail anyway.  Maybe we should just say it may have been blacklisted like before.

Reply:
Thanks for review.

I realized a fact that blacklisting actually does not work for barrier taskset. As you may know, blacklisting only takes effect when there's failed task. But for a barrier task set, it will be marked as zombie once there's any failed task and we don't consider blacklisting for a zombie task set, see:

https://github.com/apache/spark/blob/8b4862953a879a9b3ba6f57e669efc383df68b7c/core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala#L885-L894

So, I think we actually don't need to mention blacklisting at all. I will close this PR.

Reply:
+1